packages/itmat-interface/src/lxd/index.ts

```diff
@@ -0,0 +1,325 @@
+import express from 'express';
+import { WebSocketServer, WebSocket } from 'ws';
+import qs from 'qs';
+import lxdManager from './lxdManager';
+
+import jwt from 'jsonwebtoken';
+import { GraphQLError } from 'graphql';
+import { ApolloServerErrorCode } from '@apollo/server/errors';
+import { userRetrieval, userRetrievalByUserId } from '../authentication/pubkeyAuthentication';
+import { logRESTAPICall } from '../log/restLogHelper';
+
+const lxdCallsRouter = express.Router();
+
+lxdCallsRouter.use(logRESTAPICall);
+
+// Middleware for JWT verification and user retrieval, in the style of createContext
+const LXDauthenticateRequest = async (req: any, res: any, next: any) => {
+
+    const token: string = req.headers.authorization || '';
+    if ((token !== '') && (token !==undefined) && (req.user === undefined)) {
+        // skip the token start with '_xsrf'
+        if (token.startsWith('_xsrf')) {

```

Comment:
Why only filtering `_xsrf`, this is incorrect logic, please specifically target the on you need.